pyproject.toml

```diff
@@ -31,6 +31,8 @@ readability-lxml = "^0.8.1"
 requests = "^2.31.0"
 tweepy = "^4.14.0"
 "Mastodon.py" = "^1.8.1"
+lxml = {extras = ["html-clean"], version = "^5.3.0"}
+beautifulsoup4 = "^4.12.3"
```

Comment:
I'm ambivalent about this. If it's not being used by the core script, it feels like this should be codified in the specific implementation that's using it instead. Like, ideally, our internal ttnconfig repo that includes the blocklist should have a separate requirements spec that lists both ttn and bs4. (If someone else wanted to use `PyQuery` for example, they'd still have to install that themselves.)

(Looks like I actually removed this requirement in the past for similar reasons heh https://github.com/freedomofpress/trackthenews/commit/64bcb02123366ccca84339d4806b95cd5f440134)

On the other hand, this is a small project that's only used by a handful of people besides us, so I'd accept if it feels like too much bikeshedding to debate this. Probably not a big deal to include.

Reply:
What do you think about dropping an abstract base class in this repo that _does_ import bs4? Basically, more of an opinionated plugin-style design pattern intended to minimize the complexity folks may have to wrangle to add something like a basic blocklist.

Reply:
Sure, sounds reasonable enough!

Reply:
OK, I'll poke at this still in this PR, dragging back to WIP until that's done

Reply:
This is now done; I've avoided importing BeautifulSoup in the ABC since it serves no purpose there, but I've annotated `pyproject.toml` to make it clearer why it's there.